File: ex4/python/mad_libs.py
import random

context = [
    "Do you {verb} your {adjective} {noun} {adverb}",
]


def mad_libs(base, noun, verb, adjective, adverb):
    mapped = {'noun':noun, 'verb':verb, 'adjective':adjective, 'adverb':adverb}
    return base.format_map(mapped)


if __name__=="__main__":
    if len(context) is 0:
        raise ValueError("context is empty")
    _base = random.shuffle(context)
    _noun = input("Enter a noun:")
    _verb = input("Enter a verb:")
    _adjective = input("Enter a adjective:")
    _adverb = input("Enter a adverb:")
    print(mad_libs(_base, _noun, _verb, _adjective, _adverb))

File: ex4/src/main/java/codingtraining/ex4/MadLibs.java
package codingtraining.ex4;

import java.util.Scanner;

public class MadLibs {
    private static final String NOUN = "${NOUN}";
    private static final String VERB = "${VERB}";
    private static final String ADJECTIVE = "${ADJECTIVE}";
    private static final String ADVERB = "${ADVERB}";

    private static final String FORMAT = "Do you ${VERB} your ${ADJECTIVE} ${NOUN} ${ADVERB}? That's hilarious";

    public static String getMadLibs(String format, String noun, String verb, String adjective, String adverb) {
        format = format.replace(NOUN, noun);
        format = format.replace(VERB, verb);
        format = format.replace(ADJECTIVE, adjective);
        format = format.replace(ADVERB, adverb);
        return format;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a noun: ");
        String noun = scanner.nextLine();
        System.out.print("Enter a verb: ");
        String verb = scanner.nextLine();
        System.out.print("Enter a adjective: ");
        String adjective = scanner.nextLine();
        System.out.print("Enter a adverb: ");
        String adverb = scanner.nextLine();
        System.out.println(getMadLibs(FORMAT, noun, verb, adjective, adverb));
    }
}
